src/etc/inc/xmlparse.inc

```diff
@@ -44,7 +44,7 @@ function listtags()
       'member', 'menu', 'mobilekey', 'monitor_type', 'mount',
       'npt', 'ntpserver',
       'onetoone', 'openvpn-server', 'openvpn-client', 'openvpn-csc', 'option',
-      'package', 'passthrumac', 'phase1', 'phase2', 'ppp', 'pppoe', 'priv', 'proxyarpnet', 'pool', 'pages', 'pipe',
+      'package', 'passthrumac', 'phase1', 'phase2', 'ppp', 'pppoe', 'priv', 'proxyarpnet', 'pool', 'pages', 'pipe', 'publickey',
```

Comment:
Is this really necessary? The code is old and in need of replacement and the list trickery in particular is prone to errors.

Reply:
I have noticed that the whole IPsec plugin is based on legacy code, but as multiple keys (RSA public + optional private key) can be stored, I can not think of an array-less way of storing this data. I made sure to follow the current code style / implementation, specifically following along the IPsec Mobile Keys implementation. And no other free / unused arrays existed for configuring this, however I am happy to rename the key to e.g. ipsec-publickey if preferred.

Reply:
I double-checked and the only other key which seems suitable would be `key`, however this seems ambiguous to me as the IPsec plugin already makes use of `mobilekey`. I would prefer to introduce `pluginkey` as a list tag, as it does not seem to be in use anywhere in the current OPNsense core (so it should not break anything hopefully, if I understood this function correctly) and it would clearly show which kind of key it is referring to.

Reply:
Maybe we better move the new functionality to a new mvc based form, it would have the advantage of not increasing the legacy codebase and it will likely be quite easy to use the new (public key) entries in the old codebase.

Reply:
@AdSchellevis I'm more than happy to look into this as well - I have zero experience with the new MVC model but at a first glance it seems to be more sane. However I am not sure if a mixed solution is appreciated, so I'd wait for agreement / feedback from @fichtner or @fabianfrz before giving this a try.

Reply:
I do trust @AdSchellevis on this, it's a good idea to avoid the early concerns. :)

Access from legacy to MVC is quite easy:

1. Use via MVC model: https://github.com/opnsense/core/blob/master/src/etc/inc/plugins.inc.d/netflow.inc#L31-L33

2. Use via raw config access: https://github.com/opnsense/core/blob/master/src/etc/inc/plugins.inc.d/squid.inc#L35-L36

Reply:
@ppmathis Today I've added a bit of extra documentation on how to use our mvc components for these type of use cases (it was still somewhere on the list of things todo).  https://docs.opnsense.org/development/examples/using_grids.html